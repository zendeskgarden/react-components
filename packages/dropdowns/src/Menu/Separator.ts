/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import { HTMLProps } from 'react';
import { StyledSeparator } from '../styled';

/**
 * Accepts all `<li>` props
 */
const Separator = StyledSeparator as React.FunctionComponent<HTMLProps<HTMLLIElement>>;

Separator.displayName = 'Separator';

/** @component */
export default Separator;
