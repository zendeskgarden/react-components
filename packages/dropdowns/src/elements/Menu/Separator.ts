/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import { HTMLAttributes } from 'react';
import { StyledSeparator } from '../../styled';

/**
 * Accepts all `<li>` props
 */
const Separator = StyledSeparator as React.FunctionComponent<HTMLAttributes<HTMLLIElement>>;

Separator.displayName = 'Separator';

/** @component */
export default Separator;
