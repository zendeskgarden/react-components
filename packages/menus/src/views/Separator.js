import styled from 'styled-components';
import { retrieveTheme } from '@zendesk/garden-react-theming';
import MenuStyles from '@zendesk/garden-css-menus';

import { version } from '../../package.json';
const COMPONENT_ID = 'menus.separator';

/**
 * Accepts all `<li>` props
 */
const Separator = styled.li.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version,
  role: 'separator',
  className: MenuStyles['c-menu__separator']
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default Separator;
