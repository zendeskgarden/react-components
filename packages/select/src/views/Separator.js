import { retrieveTheme } from '@zendesk/garden-react-theming';
import { Separator as MenuSeparator } from '@zendesk/garden-react-menus';

import { version } from '../../package.json';
const COMPONENT_ID = 'select.separator';

/**
 * Accepts all `<li>` props
 */
const Separator = MenuSeparator.extend.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default Separator;
