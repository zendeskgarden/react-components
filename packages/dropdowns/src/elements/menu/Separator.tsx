/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { LiHTMLAttributes, forwardRef } from 'react';
import { StyledSeparator } from '../../views';
import useMenuContext from '../../context/useMenuContext';

/**
 * @extends LiHTMLAttributes<HTMLLIElement>
 */
export const Separator = forwardRef<HTMLLIElement, LiHTMLAttributes<HTMLLIElement>>(
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ({ children, ...props }, ref) => {
    const { getSeparatorProps } = useMenuContext();

    const separatorProps = getSeparatorProps() as LiHTMLAttributes<HTMLLIElement>;

    return <StyledSeparator {...props} {...separatorProps} ref={ref} />;
  }
);

Separator.displayName = 'Separator';
