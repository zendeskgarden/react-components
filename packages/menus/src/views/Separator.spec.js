import React from 'react';
import { shallow } from 'enzyme';
import Separator from './Separator';

describe('Separator', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<Separator />);

    expect(wrapper).toMatchSnapshot();
  });
});
