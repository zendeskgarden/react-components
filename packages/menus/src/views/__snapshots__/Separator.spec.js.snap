// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator renders default styling 1`] = `
<li
  className="c-menu__separator "
  data-garden-id="menus.separator"
  data-garden-version="version"
  role="separator"
/>
`;
